uint8 algorithm_number
---
bool result
